core/chelpers.c

```diff
@@ -79,7 +79,7 @@ void set_stdout(char *f, int *sid, int flen)
   if (logfile[0] != '\0') {
     redirect = 1;
   } 
-  else if (envvar = getenv("NEK_LOGFILE")) {
+  else if (envvar == getenv("NEK_LOGFILE")) {
```

Comment:
I think this block should read:
```
else if (getenv("NEK_LOGFILE")) {
    char *envvar = getenv("NEK_LOGFILE");
    if (*sid >= 0) sprintf(logfile, "s%05d_", *sid);
    strcat(logfile + strlen(logfile), envvar);
    redirect = 1;
}

```

Reply:
You're probably right, I didn't look too closely at what this code does.